src/entity/spatial/LinearRing.cpp

```diff
@@ -0,0 +1,59 @@
+#include <sempr/entity/spatial/LinearRing.hpp>
+#include <LinearRing_odb.h>
+
+namespace sempr { namespace entity {
+
+SEMPR_ENTITY_SOURCE(LinearRing)
+
+LinearRing::LinearRing() : LinearRing(new core::IDGen<LinearRing>())
+{
+}
+
+LinearRing::LinearRing(const core::IDGenBase* idgen)
+    : LineString(idgen)
+{
+    this->setDiscriminator<LinearRing>();
+    geometry_ = factory_->createLinearRing();
```

Comment:
Memory leak: The c'tor of `LineString` comes first and has already assigned a geometry at this point:
` geometry_ = factory_->createLineString();`
Now we assign a new pointer without deleting the old geometry. I guess this pattern is present in most of the geometry-classes? (I will only comment this once)

Reply:
I don't see a reason to call the superclass c'tor here.
Maybe just call the c'tor from Entity directly to get the eventbroker?

Reply:
The superclasses c'tor is called automatically, there is no way to prevent that. The explicit "call" here just hands over the `core::IDGenBase*` down to the next level, so that the c'tor of `DBObject` can use it to create an ID. This mechanism allows us to create an ID based on the most-derived type of object but have it ready in every c'tor along the inheritance chain. [1]

If we didn't specify the c'tor of `LineString` its default c'tor would be used.
If we tried to jump to another c'tor, e.g. `Entity`, we'd get a compiler error because `LinearRing` does not directly inherit from `Entity`. :wink:

[1] I guess this mechanism is subject to change (hopefully) soon. ^^

Reply:
Fixed in a13e87f2aff910ec64bf0921afa723999c77c8c6 by breaking this inheritance.